### PR TITLE
Idex 3060

### PR DESCRIPTION
**Pull request by garagatyi**

### CODE REVIEW

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java [hunk 1]
```diff
@@ -105,6 +111,7 @@ public MachineManager(SnapshotDao snapshotDao,
         this.defaultMachineMemorySizeMB = defaultMachineMemorySizeMB;
 
         executor = Executors.newCachedThreadPool(new ThreadFactoryBuilder().setNameFormat("MachineManager-%d").setDaemon(true).build());
+        this.machineCleaner = new MachineCleaner();
```
Comment (skabashnyuk) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java]: why not injected?

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java [hunk 2]
```diff
@@ -872,8 +886,32 @@ private LineConsumer getLogger(LineConsumer fileLogger, String outputChannel) th
         return fileLogger;
     }
 
+    // cleanup machine if event about instance failure comes
+    private class MachineCleaner implements EventSubscriber<InstanceStateEvent> {
```
Comment (skabashnyuk) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java]: may be separate class?

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java [hunk 3]
```diff
@@ -872,8 +886,32 @@ private LineConsumer getLogger(LineConsumer fileLogger, String outputChannel) th
...
+                    if (event.getType() == OOM) {
+                        message = message +
+                                  "The processes in this machine need more RAM. This machine started with " + machine.getMemorySize() +
+                                  "MB. Create a new machine configuration that allocates additional RAM or increase " +
+                                  "the workspace RAM limit in the user dashboard.";
+                    }
+
+                    cleanupOnDestroy(machine, message);
+                } catch (NotFoundException | MachineException ignore) {
+                }
```
Comment (skabashnyuk) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java]: Log.warn ?

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java (hunk 1 above)
Comment (garagatyi) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java]: It uses private method, so I make it private class.

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java (hunk 2 above)
Comment (garagatyi) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java]: It uses private method of outer class

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java (hunk 3 above)
Comment (garagatyi) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java]: I would say it is normal situation. Can happen when user destroy machine by API

**Comment by skabashnyuk** [comment]
other ok

**Pull request closed without merging**